pkg/network/ebpf/c/protocols/classification/protocol-classification.h

```diff
@@ -64,12 +104,26 @@ static __always_inline protocol_t classify_queue_protocols(struct __sk_buff *skb
     return PROTOCOL_UNKNOWN;
 }
 
-__maybe_unused static __always_inline void save_protocol(conn_tuple_t *skb_tup, protocol_t cur_fragment_protocol) {
-    bpf_map_update_with_telemetry(connection_protocol, skb_tup, &cur_fragment_protocol, BPF_NOEXIST);
-    conn_tuple_t inverse_skb_conn_tup;
-    bpf_memcpy(&inverse_skb_conn_tup, skb_tup, sizeof(conn_tuple_t));
-    flip_tuple(&inverse_skb_conn_tup);
-    bpf_map_update_with_telemetry(connection_protocol, &inverse_skb_conn_tup, &cur_fragment_protocol, BPF_NOEXIST);
+static __always_inline protocol_stack_t* get_protocol_stack(struct __sk_buff *skb, usm_context_t *usm_ctx) {
+    conn_tuple_t normalized_tup = usm_ctx->tuple;
+    normalize_tuple(&normalized_tup);
```

Comment:
Might be not part of the scope for this PR, but given normalize_tuple becomes a "first_class_citizen" in our codebase, maybe it is worth to perform this action as soon as possible to prevent potential problems, and assume in our entire codebase that we work with normalized tuples only.

for example, in this case maybe we should consider normalizing the tuple in `usm_context_init`

Reply:
@p-lambert @guyarb what do you think about this? i do think that we should handle the `normalize_tuple` logic in a separate PR (if at all), just want to make sure we don't lose this comment

Reply:
Makes sense to me, but I'd rather address this in a separate PR because I'm not 100% that we're not relying on the original tuple order for other things